craft_store/publishergateway/_publishergw.py

```diff
@@ -0,0 +1,107 @@
+
+    This class is a client wrapper for the Canonical Publisher Gateway.
+    The latest version of the server API can be seen at: https://api.charmhub.io/docs/
+
+    Each instance is only valid for one particular namespace.
+    """
+
+    def __init__(self, base_url: str, namespace: str, auth: auth.Auth) -> None:
+        self._namespace = namespace
+        self._client = httpx.Client(
+            base_url=base_url,
+            auth=candidauth.CandidAuth(auth=auth, auth_type="macaroon"),
+        )
+
+    @staticmethod
+    def _check_error(response: httpx.Response) -> None:
+        if response.is_success:
+            return
+        try:
+            error_response = response.json()
+        except Exception as exc:
+            raise errors.CraftStoreError(
+                f"Invalid response from server ({response.status_code})",
+                details=response.text,
+            ) from exc
+        error_list = error_response.get("error-list", [])
+        if response.status_code >= 500:
+            brief = f"Store had an error ({response.status_code})"
+        else:
+            brief = f"Error {response.status_code} returned from store"
+        if len(error_list) == 1:
+            brief = f"{brief}: {error_list[0].get('message')}"
+        else:
+            brief = f"{brief}. See log for details"
+        raise errors.CraftStoreError(
+            brief, store_errors=errors.StoreErrorList(error_list)
+        )
+
+    def get_package_metadata(self, name: str) -> _response.PackageMetadata:
+        """Get general metadata for a package.
+
+        :param name: The name of the package to query.
+        :returns: A dictionary matching the result from the publisher gateway.
+
+        API docs: https://api.charmhub.io/docs/default.html#package_metadata
+        """
+        response = self._client.get(
+            url=f"/v1/{self._namespace}/{name}",
+        )
+        self._check_error(response)
+        return cast(_response.PackageMetadata, response.json()["metadata"])
+
+    def create_tracks(self, name: str, *tracks: _request.CreateTrackRequest) -> int:
+        """Create one or more tracks in the store.
+
+        :param name: The store name (i.e. the specific charm, snap or other package)
+            to which this track will be attached.
+        :param tracks: Each track is a dictionary mapping query values.
+        :returns: The number of tracks created by the store.
+        :raises: ValueError if a track name is invalid.
+
+        API docs: https://api.charmhub.io/docs/default.html#create_tracks
+        """
+        bad_track_names = {
+            track["name"]
+            for track in tracks
+            if not _request.TRACK_NAME_REGEX.match(track["name"])
+            or len(track["name"]) > 28
+        }
+        if bad_track_names:
+            bad_tracks = ", ".join(sorted(bad_track_names))
+            raise ValueError(f"The following track names are invalid: {bad_tracks}")
```

Comment:
We may think of raising `InvalidRequestError` or something similar in here instead of raw ValueError, as it may be useful to return `error-like` structure so it will be properly displayed to the user